src/praisonai/praisonai/integrations/managed_local.py

```
@@ -460,29 +455,80 @@ def _restore_state(self) -> None:
+        # If compute provider is attached, install in sandbox
+        if self._compute and self._compute_instance_id:
+            logger.info("[local_managed] installing pip packages in sandbox: %s", pip_pkgs)
+            cmd = f"{sys.executable} -m pip install -q " + " ".join(pip_pkgs)
             try:
-                subprocess.run(cmd, check=True, capture_output=True, timeout=120)
-            except subprocess.CalledProcessError as e:
-                logger.warning("[local_managed] pip install failed: %s", e.stderr)
-            except subprocess.TimeoutExpired:
-                logger.warning("[local_managed] pip install timed out")
+                await self._compute.execute(self._compute_instance_id, cmd, timeout=120)
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)